contracts/query-queue/tests/integration.rs

```diff
@@ -0,0 +1,215 @@
+//! Then running `cargo integration-test` will validate we can properly call into that generated Wasm.
+//!
+//! You can easily convert unit tests to integration tests as follows:
+//! 1. Copy them over verbatim
+//! 2. Then change
+//!      let mut deps = mock_dependencies(20, &[]);
+//!    to
+//!      let mut deps = mock_instance(WASM, &[]);
+//! 3. If you access raw storage, where ever you see something like:
+//!      deps.storage.get(CONFIG_KEY).expect("no data stored");
+//!    replace it with:
+//!      deps.with_storage(|store| {
+//!          let data = store.get(CONFIG_KEY).expect("no data stored");
+//!          //...
+//!      });
+//! 4. Anywhere you see query(&deps, ...) you must replace it with query(&mut deps, ...)
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{
+    from_binary, from_slice, to_binary, Binary, ContractResult, MessageInfo, Response, SystemError,
+    SystemResult, WasmQuery,
+};
+use cosmwasm_vm::{
+    testing::{
+        execute, instantiate, mock_env, mock_info, mock_instance_with_gas_limit, query, MockApi,
+        MockQuerier, MockStorage,
+    },
+    Backend, Instance, InstanceOptions, Storage, VmResult,
+};
+
+use query_queue::contract::{RawResponse, SumResponse};
+use query_queue::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
+
+static WASM: &[u8] = include_bytes!("../target/wasm32-unknown-unknown/release/query_queue.wasm");
+static QUEUE_WASM: &[u8] =
+    include_bytes!("../../queue/target/wasm32-unknown-unknown/release/queue.wasm");
+
+fn create_contract() -> (Instance<MockApi, MockStorage, MockQuerier>, MessageInfo) {
+    let gas_limit = 500_000_000; // enough for many executions within one instance
+    let instance_options = InstanceOptions {
+        gas_limit,
+        print_debug: false,
+    };
+    let mut deps = Backend {
+        api: MockApi::default(),
+        storage: MockStorage::new(),
+        querier: MockQuerier::new(&[]),
+    };
+    let info = mock_info("creator", &[]);
+    deps.querier.update_wasm(|query_msg| match query_msg {
+        WasmQuery::Smart { contract_addr, msg } => {
+            if contract_addr != "queue_address" {
+                return SystemResult::Err(SystemError::NoSuchContract {
+                    addr: contract_addr.to_string(),
+                });
+            };
+            let q_msg: QueryMsg = from_slice(msg).unwrap();
+            match q_msg {
+                QueryMsg::Sum {} => SystemResult::Ok(ContractResult::Ok(
+                    to_binary(&SumResponse { sum: 42 }).unwrap(),
+                )),
+                _ => SystemResult::Err(SystemError::Unknown {}),
+            }
+        }
+        _ => SystemResult::Err(SystemError::Unknown {}),
+    });
+    let mut instance = Instance::from_code(WASM, deps, instance_options, None).unwrap();
+    let res: Response = instantiate(
+        &mut instance,
+        mock_env(),
+        info.clone(),
```

Comment:
It was changed to `info` from `info.clone()` in the ccontract.rs's unit-test.
But this code use `clone()` function. Is it no problem?

https://github.com/line/cosmwasm/pull/197/files#diff-08133943a6c9a670f2c1833b7eb386f365d546acbcd5386ccab8d649dd37783cR155